cylc/flow/task_job_mgr.py

```diff
@@ -1125,7 +1125,9 @@ def _prep_submit_task_job(
                 )
             else:
                 platform_name = self.task_remote_mgr.subshell_eval(
-                    rtconfig['platform'], PLATFORM_REC_COMMAND
+                    rtconfig['platform'],
+                    PLATFORM_REC_COMMAND,
+                    host_check=False
```

Comment:
I'm concerned this will have unintended effects.

Does the bug not actually lie here? https://github.com/cylc/cylc-flow/blob/79e324df6519d1a007165e4878dd325dd7a1c4a5/cylc/flow/hostuserutil.py#L192-L200

Reply:
Ok makes sense. The `subshell_eval` method is somewhat confusing to figure out.

Also I'm still wondering about https://github.com/cylc/cylc-flow/blob/79e324df6519d1a007165e4878dd325dd7a1c4a5/cylc/flow/hostuserutil.py#L200

If "localhost.localdomain[.whatever]" is the only correct pattern then it should be
```diff
-name.split(".")[0].startswith("localhost")
+name.split(".")[0] == "localhost"
```

Reply:
cc @oliver-sanders

Reply:
There can be many different variants of `localhost` e.g. `localhost.localdomain` etc, which is why the test is `startswith('localhost')` rather than `== 'localhost'`.

Reply:
I'm confused by the `host_check` stuff in the first place though - https://github.com/cylc/cylc-flow/pull/5343#issuecomment-1433059203

Reply:
> There can be many different variants of `localhost` e.g. `localhost.localdomain` etc, which is why the test is `startswith('localhost')` rather than `== 'localhost'`.

But I'm asking if those variants will always follow the pattern of `locahost.something`. Because if not, then the `.split(".")[0]` is pointless

Reply:
I too am curious - is `localhost-foo.localdomain` a real possibility - and if it is, do we want it to match?

Reply:
I think this is a possibility, looking at /etc/hosts file where a similar pattern is used seems like we should cater for it?

Reply:
I think we had some examples of cases where the logic _is_ necessary - can someone remind me what they are, and I'll add a comment to this to avoid this debate happening again?

Reply:
My suggestion to avoid future confusion
```diff
-if not name or name.split(".")[0].startswith("localhost"): 
-    # e.g. localhost.localdomain 
+if not name or name.startswith("localhost"):
+    # e.g. localhost4.localdomain4
```
and actually unit test this function!